### PR TITLE
Add an alert to detect missing successful results from heartbeat jobs.

### PR DESCRIPTION
**Pull request by cjwagner**
/hold
Holding as this is a follow up to https://github.com/kubernetes/test-infra/pull/19470 (The first commit is from that PR)

### CODE REVIEW

File: config/prow/cluster/monitoring/mixins/prometheus/plank_alerts.libsonnet [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  prometheusAlerts+:: {
+    local componentName = $._config.components.plank,
+    groups+: [
+      {
+        name: 'Heartbeat ProwJobs',
+        rules: [
+          {
+            alert: 'No recent successful runs: `%s`' % job.name,
+            expr: |||
+              increase(prowjob_state_transitions{job_name="%s", state="success"}[%s]) < 0.5
```
Comment (e-blackwelder) [inline config/prow/cluster/monitoring/mixins/prometheus/plank_alerts.libsonnet:11]: Please add a comment explaining what this does.

Comment (cjwagner) [reply config/prow/cluster/monitoring/mixins/prometheus/plank_alerts.libsonnet:20]: Done.

**Review by e-blackwelder — APPROVED** [review]

**Comment by cjwagner** [comment]
The value of `increase(prowjob_state_transitions{job_name="ci-test-infra-prow-checkconfig", state="success"}[20m]) ` is dropping down to 1 occasionally due to the alertInterval being exactly double the job interval. I'm going to decrease the job interval to 9 mins to avoid this without increasing the alert interval. As is we might accidentally fire an alert on a single transient failure.
Screen Shot 2020-10-07 at 1 58 28 PM

**Review by e-blackwelder — APPROVED** [review]

**Pull request merged**